pkg/gcp/machine_controller_util.go

```diff
@@ -83,34 +83,32 @@ func (ms *MachinePlugin) CreateMachineUtil(ctx context.Context, machineName stri
 		instance.Description = *providerSpec.Description
 	}
 
-	disks := []*compute.AttachedDisk{}
+	var disks []*compute.AttachedDisk
 	for _, disk := range providerSpec.Disks {
-		var attachedDisk compute.AttachedDisk
-		autoDelete := false
-		if disk.AutoDelete == nil || *disk.AutoDelete == true {
-			autoDelete = true
+		attachedDisk := compute.AttachedDisk{
+			Type:       validation.DiskTypePersistent,
```

Comment:
Isn't it better as a general case to pass the `disk.Type` here that you anyway get from the input, rather than "assuming" the PERSISTENT ?

Reply:
There are two types of disk type.
The actual disk type has two possible values: "PERSISTENT" and "SCRATCH".
The disk type we pass is the `AttachedDiskInitializeParams.DiskType`.